py/desitarget/streams/cuts.py

```diff
@@ -0,0 +1,328 @@
+
+    # ADM look up the defining parameters of the stream.
+    stream = get_stream_parameters(stream_name)
+    # ADM the parameters that define the coordinates of the stream.
+    rapol, decpol, ra_ref = stream["RAPOL"], stream["DECPOL"], stream["RA_REF"]
+    # ADM the parameters that define the extent of the stream.
+    mind, maxd = stream["MIND"], stream["MAXD"]
+
+    # ADM rotate the data into the coordinate system of the stream.
+    fi1, fi2 = sphere_rotate(objs['RA'], objs['DEC'], rapol, decpol, ra_ref)
+
+    # ADM distance of the stream (similar to Koposov et al. 2010 paper).
+    dist = stream_distance(fi1, stream_name)
+
+    # ADM heliocentric correction to proper motion.
+    xpmra, xpmdec = correct_pm(objs['RA'], objs['DEC'],
+                               objs['PMRA'], objs['PMDEC'], dist)
+
+    # ADM reflex correction for proper motions.
+    pmfi1, pmfi2 = rotate_pm(objs['RA'], objs['DEC'], xpmra, xpmdec,
+                             rapol, decpol, ra_ref)
+
+    # ADM derive the combined proper motion error.
+    # ADM guard against dividing by zero.
+    pmra_error = np.zeros_like(objs["PMRA_IVAR"]) + 1e8
+    ii = objs['PMRA_IVAR'] != 0
+    pmra_error[ii] = 1./np.sqrt(objs[ii]['PMRA_IVAR'])
+    pmdec_error = np.zeros_like(objs["PMDEC_IVAR"]) + 1e8
+    ii = objs['PMDEC_IVAR'] != 0
+    pmdec_error[ii] = 1./np.sqrt(objs[ii]['PMDEC_IVAR'])
+    pm_err = np.sqrt(0.5 * (pmra_error**2 + pmdec_error**2))
+
+    # ADM dust correction.
+    ext_coeff = dict(g=3.237, r=2.176, z=1.217)
+    eg, er, ez = [ext_coeff[_] * objs['EBV'] for _ in 'grz']
+    ext = {}
+    ext['G'] = eg
+    ext['R'] = er
+    ext['Z'] = ez
+
+    g, r, z = [22.5 - 2.5 * np.log10(objs['FLUX_' + _]) - ext[_] for _ in 'GRZ']
+
+    # ADM some spline function over which to interpolate.
+    TRACK = scipy.interpolate.CubicSpline([-90, -70, -50, -40, -20, 0, 20],
+                                          [-3, -1.5, -.2, -0., -.0, -1.2, -3])
+    PM1TRACK = scipy.interpolate.UnivariateSpline(
+        [-90, -70, -50, -30, -15, 0, 20],
+        [-5.5, -6.3, -8, -6.5, -5.7, -5, -3.5])
+    PM2TRACK = scipy.interpolate.UnivariateSpline(
+        [-90, -60, -45, -30, 0, 20],
+        [-.7, -.7, -0.35, -0.1, 0.2, .5], s=0)
+
+    # ADM create an interpolated set of phi2 coords (in stream coords).
+    dfi2 = fi2 - TRACK(fi1)
+
+    # ADM derive the isochrone track for the stream.
+    CMD_II = get_CMD_interpolator(stream_name)
+
+    # ADM how far the data lies from the isochrone.
+    delta_cmd = g - r - CMD_II(r - 5 * np.log10(dist * 1e3) + 5)
+
+    # ADM necessary parameters are set up; perform the actual selection.
+    bright_limit, faint_limit = 16, 21
+
+    # ADM lies in the stream.
+    field_sel = betw(dfi2, -10, 10) & betw(fi1, -90, 21)
+
+    # ADM Gaia-based selection (proper motion and parallax).
+    pm_pad = 2  # mas/yr padding in pm selection
+    gaia_astrom_sel = pm12_sel_func(fi1, pmfi1, pmfi2, pm_err, pm_pad, 2.5)
```

Comment:
This is not correct given the function signature
pm12_sel_func(pm1track, pm2track, pmfi1, pmfi2, pm_err, pad=2, mult=2.5)
So it should be called  something like 
pm12_sel_func(PM1TRACK(fi1), PM2TRACK(fi1), pmfi1, pmfi2, pm_err, pad=2, mult=2.5)

Reply:
I changed this to the suggested `pm12_sel_func(PM1TRACK(fi1), PM2TRACK(fi1), pmfi1, pmfi2, pm_err, pad=pm_pad, mult=2.5)`